stat_test.go

```diff
@@ -1264,6 +1265,65 @@ func TestSortWeighted(t *testing.T) {
 	}
 }
 
+func TestSortWeightedLabelled(t *testing.T) {
+	for i, test := range []struct {
+		x    []float64
+		l    []bool
+		w    []float64
+		ansx []float64
+		ansl []bool
+		answ []float64
+	}{
+		{
+			x:    []float64{8, 3, 7, 8, 4},
+			ansx: []float64{3, 4, 7, 8, 8},
+		},
+		{
+			x:    []float64{8, 3, 7, 8, 4},
+			w:    []float64{.5, 1, 1, .5, 1},
+			ansx: []float64{3, 4, 7, 8, 8},
+			answ: []float64{1, 1, 1, .5, .5},
+		},
+		{
+			x:    []float64{8, 3, 7, 8, 4},
+			l:    []bool{false, false, true, false, true},
+			ansx: []float64{3, 4, 7, 8, 8},
+			ansl: []bool{false, true, true, false, false},
+		},
+		{
+			x:    []float64{8, 3, 7, 8, 4},
+			l:    []bool{false, false, true, false, true},
+			w:    []float64{.5, 1, 1, .5, 1},
+			ansx: []float64{3, 4, 7, 8, 8},
+			ansl: []bool{false, true, true, false, false},
+			answ: []float64{1, 1, 1, .5, .5},
+		},
+	} {
+		SortWeightedLabelled(test.x, test.l, test.w)
+		if !floats.Same(test.x, test.ansx) {
+			t.Errorf("SortWeightedLabelled mismatch case %d. Expected x %v, Found x %v", i, test.ansx, test.x)
+		}
+		if (test.l != nil) && !reflect.DeepEqual(test.l, test.ansl) {
```

Comment:
Yeah that makes sense, thanks @kortschak. So am I doing the right thing using reflect.DeepEqual here as bools can't be NaN?